internal/testrunner/runners/system/system.go

```diff
@@ -3,29 +3,59 @@ package system
 import (
 	"fmt"
 
+	"github.com/pkg/errors"
+
 	"github.com/elastic/elastic-package/internal/testrunner"
+	"github.com/elastic/elastic-package/internal/testrunner/runners/system/service"
 )
 
+func init() {
+	testrunner.RegisterRunner(TestType, Run)
+}
+
 const (
 	// TestType defining system tests
 	TestType testrunner.TestType = "system"
 )
 
 type runner struct {
-	testFolderPath string
+	testFolderPath  string
+	packageRootPath string
 }
 
 // Run runs the system tests defined under the given folder
-func Run(testFolderPath string) error {
-	r := runner{testFolderPath}
+func Run(testFolderPath, packageRootPath string) error {
+	r := runner{
+		testFolderPath,
+		packageRootPath,
+	}
 	return r.run()
 }
 
 func (r *runner) run() error {
 	fmt.Println("system run", r.testFolderPath)
-	return nil
-}
 
-func init() {
-	testrunner.RegisterRunner(TestType, Run)
+	serviceRunner, err := service.Factory(r.packageRootPath)
+	if err != nil {
+		return errors.Wrap(err, "could not create service runner")
+	}
+
+	// Step 1. Setup service.
+	// Step 1a. (Deferred) Tear down service.
+	ctxt, err := serviceRunner.SetUp()
+	defer serviceRunner.TearDown(ctxt)
+	if err != nil {
+		return errors.Wrap(err, "could not setup service")
+	}
+
+	// Step 2. Setup test cluster (ES + Kibana + Agent).
+	// Step 2a. Register Agent with Fleet.
```

Comment:
@mtojek I would like your advice here: what would be the best way to accomplish these two steps (2. and 2a.) here?

Reply:
Step 2a:

It depends where the agent will be spawned. If it runs in a container then the enrollment should be performed automatically (it's in agent's Dockerfile).
I believe you have to interact with cluster to attach new container with observed product (like mysql or nginx) to the existing network, and detach it after tests.

Step 2:

I think you should assume that the cluster is already up and running, and the Agent in subscribed. You can pick up configuration options through environment variables (shellinit). I assume that the Agent is long-running and we just switch configuration inside.

Reply:
> I think you should assume that the cluster is already up and running, and the Agent in subscribed. You can pick up configuration options through environment variables (shellinit). I assume that the Agent is long-running and we just switch configuration inside.

This is the part I am most interested in. Is there some way I can _ensure_ that the cluster is up, rather than assuming it is? Are there any functions/methods I could call from here to make that happen?

> I believe you have to interact with cluster to attach new container with observed product (like mysql or nginx) to the existing network, and detach it after tests.

Cool, this is along the lines of what I was hoping for. For this I would need to know the name of the network. Is there some function/method I could call from here to get that (related to the previous paragraph, I suppose)?

Reply:
> This is the part I am most interested in. Is there some way I can ensure that the cluster is up, rather than assuming it is? Are there any functions/methods I could call from here to make that happen idempotently?

I think it' s `cluster.BootUp()`, the only issue we need to figure out is waiting for Kibana. I believe you start with the assumption that the cluster is up and fail in case it's not.

>  For this I would need to know the name of the network. Is there some function/method I could call from here to get that (related to the previous paragraph, I suppose)?

There is no such function at the moment, as it wasn't required, but I suppose we can enforce any name store it in `.env` file in `.elastic-package`.